Assets/MRTK/Providers/Windows/WindowsSpeechInputProvider.cs

```diff
@@ -244,13 +252,15 @@ private void OnPhraseRecognized(ConfidenceLevel confidence, TimeSpan phraseDurat
         {
             using (OnPhraseRecognizedPerfMarker.Auto())
             {
-                for (int i = 0; i < Commands?.Length; i++)
+                SpeechCommands[] commands = Commands;
```

Comment:
Not sure I understand why we create a local copy of the commands here.

Reply:
This is an interesting one! `Commands` is a property, which queries `CoreServices` for the input system profile to retrieve this array. Because we were using the property directly the query its length for the end condition of the `for` loop...it was getting expensive the more commands you had! Two calls per command per frame... (screenshot from a project with 65 speech commands)

![image](https://user-images.githubusercontent.com/3580640/169598973-14fee337-f37f-465f-9af1-e2a5886feb30.png)